### ISSUE TITLE
Addon impacts performance of secondary Proxmox server

### ISSUE THREAD

**Issue by londondev77** [lavinir/hassio-onedrive-backup#141]
I'm really not sure if this is a OneDrive Backup issue, a Proxmox issue or a hardware issue.

I have an old Intel NUC on which I'm running Proxmox. Home Assistant is on one virtual machine and I am running a very light-weight Ubuntu LXC on a second machine.  This Ubuntu machine is hosting a SAMBA file system and is used to stream video files that are played back on an external media player.  The drive I'm using is an SSD.

My Home Assistant instance isn't very resource intensive at all. Generally, CPU usage hovers around 10% of what is allocated to it and memory usage 70%.

I have OneDrive Backup set to do one backup a day. My Home Assistant creates two camera snapshots (only 30-40KB) an hour and these are synced using the OneDrive Backup filesync feature.

Whenever I try to watch media from my Ubuntu machine, playback at times pauses for a few seconds and then resumes.  However, if I turn off OneDrive Backup, playback is absolutely fine.  I assume it's the File Sync feature that is causing this (despite it only syncing a couple of tiny files) however when I check both virtual machines' resource usage, there are no spikes at all.

I've now set my OneDrive Backup to only run during the night and this has solved the problem.  It's not really an issue that it won't be running during the day but I'm posting this just in case other people have the same issue and it's something that can be fixed.

**Comment by lavinir** [comment]
If you see no spikes in resource usage what made you even think about disabling the addon ? :)
When you say you configured it to run only at night - do you mean by using the Allowed-Hours property ? And if so did you exempt the file sync from running during these hours ? 

Here's what the addon is doing:
* Every 5 minutes it wakes up and checks if it needs to do something - It checks the backups and the files for syncing (assuming we are inside the allowed hours).
* For backups - it checks if a new backup needs to be created, creates one if necessary and syncs to OneDrive. With normal settings you are probably creating at most one or two backups a day so I doubt this should tax your system any.
* For File Syncs - It enumerates all the files based on the paths you configured. Then it checks if the files are synced to OneDrive and copies any new / changed files. You said only about 2 files an hour .. again seems less likely. **However**, there is something  else it is doing if you enabled the option to delete removed files from OneDrive. It recursively traverses your OneDrive FileSync folder and checks if each file should be removed or not. 

Still, I can't imagine why any of this would cause high cpu / network usage based on your scenario. You could try to disable the file deletions first see if it makes a difference. Then maybe disable File Sync and keep the backups regularly with allowed hours running throughout the day and see if you notice any different behavior.

**Comment by londondev77** [comment]
Thanks for getting back to me.

> If you see no spikes in resource usage what made you even think about disabling the addon ? :) 

Call it intuition!  

> When you say you configured it to run only at night - do you mean by using the Allowed-Hours property ? And if so did you exempt the file sync from running during these hours ?

Yes, I have set the Allowed-Hours to 12-7 which is outside when I watch anything.  It also stops file syncing in that period as I hadn't enabled "Ignore allowed hours for File Sync".  Since I put this setting on, the video playback hasn't paused once with multiple videos watched.

> Still, I can't imagine why any of this would cause high cpu / network usage based on your scenario. You could try to disable the file deletions first see if it makes a difference. Then maybe disable File Sync and keep the backups regularly with allowed hours running throughout the day and see if you notice any different behavior.

I don't have Remove deleted files during File Sync enabled.

I did some further testing yesterday. I enabled the "Ignore allowed hours for File Sync" option and then played back a video.  The playback paused briefly at one point.  Peversely though, I then dropped much larger files than those normally synced into the media folder during playback. Not once did it pause so I'm at a loss.  Can we keep this ticket open so I can continue to test a bit more?  I don't have a huge amount to play back at the moment so my testing will be intermittant.

**Comment by lavinir** [comment]
Yes sure no problem, keep it open

**Comment by londondev77** [comment]
I've managed to do some limited further testing. I removed all the folders in the media folder and just managed to watch something without any freezing.  Previously, there were 28 folders, each with 25 files in them.  So I'm guessing if the filesync has to run through so many folders and files to detect changes, maybe this was causing the performance issues?

Personally, I'm fine limiting the addon to only work in certain hours but thought I'd start this thread in case it helps someone else.  Not sure if there is anything that could help performance.

**Comment by londondev77** [comment]
I think the SSD is the culprit here. My server blew after a power cut along with the SSD. I replaced the server with the same model but bought a different brand of SSD and since then, there have been no freezes during video playback. 

Thanks for your responses on this issue and I think this can be closed.

**Comment by londondev77** [comment]
Reopening this as unfortunately, this issue has started again. I assume no one else is having this issue so feel free to close it if you don't think it's worth investigating further.  I'll just set my sync to only happen during the night which I assume will continue to prevent the issues reported above.

**Comment by londondev77** [comment]
Adding some more info to this in case it helps identify what the issue is.

I have recently started using the Cloudflared addon. It works well although I noticed the tunnel kept failing and restarting.  When I turned off the filesync feature of the OneDrive backup addon, the tunnel issues went away and it stayed up permanently.

I've looked at the logs for OneDrive backup and there seem to be errors around the time the Cloudflare tunnel drops.  Here are some of the log entries:
```
05/21/2024 11:38:56 [13] Error: Error syncing file /media/Blink-cameras/side/20240515/11.jpg: System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.HeadersInspectionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.CompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.DriveItemItemRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.GetItemInAppFolderAsync(String subPath) in /src/Graph/GraphHelper.cs:line 121
   at hassio_onedrive_backup.Sync.SyncManager.SyncFile(String path) in /src/Sync/SyncManager.cs:line 170
   at hassio_onedrive_backup.Sync.SyncManager.SyncLoop(CancellationToken ct) in /src/Sync/SyncManager.cs:line 110
   
05/21/2024 11:51:12 [3] Error: Error syncing file /media/Blink-cameras/side/20240518/09.jpg: System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.HeadersInspectionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.CompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.DriveItemItemRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.GetItemInAppFolderAsync(String subPath) in /src/Graph/GraphHelper.cs:line 121
   at hassio_onedrive_backup.Sync.SyncManager.SyncFile(String path) in /src/Sync/SyncManager.cs:line 170
   at hassio_onedrive_backup.Sync.SyncManager.SyncLoop(CancellationToken ct) in /src/Sync/SyncManager.cs:line 110

05/21/2024 12:01:52 [37] Error: Error syncing file /media/Blink-cameras/side/20240508/11.jpg: System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.HeadersInspectionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.CompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.DriveItemItemRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.GetItemInAppFolderAsync(String subPath) in /src/Graph/GraphHelper.cs:line 121
   at hassio_onedrive_backup.Sync.SyncManager.SyncFile(String path) in /src/Sync/SyncManager.cs:line 170
   at hassio_onedrive_backup.Sync.SyncManager.SyncLoop(CancellationToken ct) in /src/Sync/SyncManager.cs:line 110
   
05/21/2024 11:51:12 [3] Error: Error syncing file /media/Blink-cameras/side/20240518/09.jpg: System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.HeadersInspectionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.CompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.DriveItemItemRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.GetItemInAppFolderAsync(String subPath) in /src/Graph/GraphHelper.cs:line 121
   at hassio_onedrive_backup.Sync.SyncManager.SyncFile(String path) in /src/Sync/SyncManager.cs:line 170
   at hassio_onedrive_backup.Sync.SyncManager.SyncLoop(CancellationToken ct) in /src/Sync/SyncManager.cs:line 110

05/21/2024 12:01:52 [37] Error: Error syncing file /media/Blink-cameras/side/20240508/11.jpg: System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer.
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.HeadersInspectionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.Middleware.CompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.GetHttpResponseMessage(RequestInformation requestInfo, CancellationToken cancellationToken, Activity activityForAttributes, String claims, Boolean isStreamResponse)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Graph.Drives.Item.Special.Item.DriveItemItemRequestBuilder.GetAsync(Action`1 requestConfiguration, CancellationToken cancellationToken)
   at hassio_onedrive_backup.Graph.GraphHelper.GetItemInAppFolderAsync(String subPath) in /src/Graph/GraphHelper.cs:line 121
   at hassio_onedrive_backup.Sync.SyncManager.SyncFile(String path) in /src/Sync/SyncManager.cs:line 170
   at hassio_onedrive_backup.Sync.SyncManager.SyncLoop(CancellationToken ct) in /src/Sync/SyncManager.cs:line 110
```

I'm not sure what these errors are showing, if they are causing the Cloudflare tunnel to go down (and the streaming issue I opened this ticket about) or whether they can be fixed but it would good to know if there are any steps I can take.

Thanks for your help!

**Comment by londondev77** [comment]
A little more info:
I was syncing 28 folders each containing 25 files so I thought maybe the constant scanning of files was having too much of a performance hit. So I reduced this to 4 folders with 25 files.  I didn't see errors in the logs but the tunnel kept dropping.

I did try increasing the number of cores and sockets devoted to the VM running Home Assistant to no avail. I never saw a peak of more than about 10% in the CPU usage so I'm pretty sure there's more than enough compute power.  Memory usage never peaked either.

So I've now stopped filesyncing completely and the tunnel hasn't dropped once.
